python/rikai/types/vision.py

```diff
@@ -43,15 +43,20 @@ class Image(ToNumpy, ToPIL, Asset, Displayable):
 
     Parameters
     ----------
-    uri : str
+    data : bytes, optional
+        The content of the image.
+    uri : str or Path, optional
         The URI pointed to an Image stored on external storage.
     """
 
     __UDT__ = ImageType()
 
-    def __init__(self, uri: Union[str, Path]):
-        super().__init__(uri)
-        self._cached_data = None
+    def __init__(
+        self,
+        data: Optional[bytes] = None,
```

Comment:
will creating images from embedded data be more common than creating images from uri's? If not maybe swap the order so you don't need to explicitly specify the parameter name?

Another thing to consider might be that while it makes sense for the Asset ABC to separate these parameters, maybe the "user-facing" classes like `Image` can just use a single parameter (non-option) and figure out whether it's the data or the uri based on the input type?

Reply:
good point.  let me update this PR>